src/sql/src/parsers/create_parser.rs

```diff
@@ -253,74 +253,160 @@ impl<'a> ParserContext<'a> {
         columns: &mut Vec<ColumnDef>,
         constraints: &mut Vec<TableConstraint>,
     ) -> Result<()> {
-        let column = self
-            .parser
+        let mut column = self
             .parse_column_def()
             .context(SyntaxSnafu { sql: self.sql })?;
 
-        if !matches!(
-            column.data_type,
-            DataType::Timestamp(_, _) | DataType::BigInt(_)
-        ) || matches!(self.parser.peek_token(), Token::Comma)
-        {
-            columns.push(column);
-            return Ok(());
-        }
+        let mut time_index_opt_idx = None;
+        for (index, opt) in column.options.iter().enumerate() {
+            if let ColumnOption::DialectSpecific(tokens) = &opt.option {
+                if matches!(
+                    &tokens[..],
+                    [
+                        Token::Word(Word {
+                            keyword: Keyword::TIME,
+                            ..
+                        }),
+                        Token::Word(Word {
+                            keyword: Keyword::INDEX,
+                            ..
+                        })
+                    ]
+                ) {
+                    let constraint = TableConstraint::Unique {
+                        name: Some(Ident {
+                            value: TIME_INDEX.to_owned(),
+                            quote_style: None,
+                        }),
+                        columns: vec![Ident {
+                            value: column.name.value.clone(),
+                            quote_style: None,
+                        }],
+                        is_primary: false,
+                    };
 
-        // for supporting `ts TIMESTAMP TIME INDEX,` syntax.
-        self.parse_time_index(column, columns, constraints)
-    }
+                    constraints.push(constraint);
+                    time_index_opt_idx = Some(index);
+                }
+            }
+        }
 
-    fn parse_time_index(
-        &mut self,
-        mut column: ColumnDef,
-        columns: &mut Vec<ColumnDef>,
-        constraints: &mut Vec<TableConstraint>,
-    ) -> Result<()> {
-        self.parser
-            .expect_keywords(&[Keyword::TIME, Keyword::INDEX])
-            .context(error::UnexpectedSnafu {
-                sql: self.sql,
-                expected: "TIME INDEX",
-                actual: self.peek_token_as_string(),
-            })?;
+        if let Some(index) = time_index_opt_idx {
```

Comment:
return an error on `else`?

Reply:
No ,it can't. Because the `time index` may exist in table constraints. And it will be checked at runtime.